google/datalab/bigquery/_query.py

```diff
@@ -45,32 +45,35 @@ def __init__(self, sql, values=None, udfs=None, data_sources=None,
           It is possible to have variable references in a query string too provided the variables
           are passed as keyword arguments to this constructor.
 
-      values: a dictionary used to expand variables if passed a SqlStatement or a string with
-          variable references.
+      env: a dictionary containing objects from the query execution context, used to get references
+          to UDFs, subqueries, and external data sources referenced by the query
+      query_params: a dictionary containing query parameter types and values, passed to BigQuery.
       udfs: list of UDFs referenced in the SQL.
       data_sources: dictionary of external data sources referenced in the SQL.
       subqueries: list of subqueries referenced in the SQL
 
     Raises:
       Exception if expansion of any variables failed.
       """
+    self._sql = sql
     self._data_sources = data_sources
     self._udfs = udfs
     self._subqueries = subqueries
-    self._values = values
+    self._env = env
+    self._query_params = query_params
 
     if data_sources is None:
       data_sources = {}
 
     self._code = None
     self._imports = []
-    if self._values is None:
-      self._values = {}
-
-    self._sql = google.datalab.data.SqlModule.expand(sql, self._values)
```

Comment:
It looks like the tests rely on this method being called in order to convert instances of SqlStatement to strings.

If we get rid of it, then we still need to do that conversion.